### ISSUE TITLE
SQL length constraints not enforced in data of noisy and bad qualities

### ISSUE THREAD

**Issue by diegoquintanav** [diegoquintanav/acme-data-generation#2]
- FlightID in table flights with a quality other than good produces strings with char lengths incompatible with the SQL tables 
- bad quality delay codes are 3 chars long when the max is CHAR(2)

these problems need more inspection, and should be fixed with a test

**Comment by diegoquintanav** [comment]
The problem lies in the columns that have the `ENUM` type in the SQL tables. It enforces a data validation that does not accept any form of noisy and bad values, which is expected. There are two ways to go around this

1. Make SQL table schema less strict, e.g. use `VARCHAR` instead of `ENUM` 
2. Make the CSV generator more strict e.g. make qualities equivalent `noisy == bad == good`

I will implement the second approach by now, that is, the `Provider` will assume produce only good quality data for those entities that use `ENUM`.

**Comment by petarjov** [comment]
The kind of a second solution is what we need but not that they are completely equal noisy == bad == good, just that they all fulfill the length constraints but they can be invalid within these constraints. 
For example delay code need to be within these values https://en.wikipedia.org/wiki/IATA_delay_codes
or you can as well consider that some chars are letters instead of digits. 

Anyhow, let me try first this version and i will let you know. 

Thanks a lot!

Best, 
Petar

**Comment by diegoquintanav** [comment]
The thing with some attributes or columns that are of ENUM based types is that they are strict on the set, e.g. for the `FrequencyUnitsKind` is

```sql
CREATE TYPE FrequencyUnitsKind AS ENUM ('Flights', 'Days', 'Miles');
```

This type is strict and it will not accept lowercase variants like `flights` or `fliGhts`, for instance. If you know of a way to produce noisy values for these attributes while keeping the ENUM type, then let me know. Otherwise it is not possible and the only way to go is by doing noisy == bad == good.

**Comment by petarjov** [comment]
For ENUM there is no need to make bad or noisy entries, so we can keep noisy == bad == good.

**Comment by diegoquintanav** [comment]
Alright, I see that you commented earlier about the IATA codes. Those can be fixed and were fixed in d9c2030, I think.
